packages/ubuntu_bootstrap/lib/pages/autoinstall/autoinstall_page.dart

```diff
@@ -1,106 +1,273 @@
-                  const SizedBox(height: kWizardSpacing),
-                  TextFormField(
-                    initialValue: model.url,
-                    onChanged: (value) => model.url = value,
-                    maxLines: null,
-                    autovalidateMode: AutovalidateMode.onUserInteraction,
-                    validator: (_) => model.state.maybeWhen(
-                      error: (error, _) {
-                        return switch (error) {
-                          YamlException _ => 'Invalid YAML',
-                          final SocketException e =>
-                            'Network error: ${e.message}',
-                          final ArgumentError e => 'Invalid URL: ${e.message}',
-                          final FormatException e =>
-                            'Invalid Format: ${e.message}',
-                          final FileSystemException e =>
-                            'File system error: ${e.message}',
-                          _ => 'Unknown Error',
-                        };
+    return Semantics(
+      label: 'Installation method selection page',
+      child: HorizontalPage(
+        windowTitle: lang.autoinstallTitle,
+        title: lang.autoinstallHeader(flavor.displayName),
+        bottomBar: WizardBar(
+          leading: Semantics(
+            button: true,
+            label: 'Back button',
+            child: const BackWizardButton(),
+          ),
+          trailing: [
+            model.autoinstall
+                ? Focus(
+                    focusNode: _validateButtonFocusNode,
+                    child: Semantics(
+                      button: true,
+                      label: 'Validate button ${model.state.isLoading ? "validating" : ""}',
+                      enabled: !model.state.hasError && model.url.isNotEmpty,
+                      child: _ValidateButton(model: model),
+                    ),
+                  )
+                : Semantics(
+                    button: true,
+                    label: 'Next button',
+                    child: WizardButton(
+                      label: UbuntuLocalizations.of(context).nextLabel,
+                      onActivated: () {
+                        SemanticsService.announce(
+                          'Proceeding to next page',
+                          TextDirection.ltr,
+                        );
+                        Wizard.of(context).next();
                       },
-                      orElse: () => null,
                     ),
                   ),
-                ],
+          ],
+        ),
+        children: [
+          // Interactive Option
+          Focus(
+            focusNode: _interactiveFocusNode,
+            onFocusChange: (hasFocus) {
+              if (hasFocus) {
+                SemanticsService.announce(
+                  'Interactive installation option ${!model.autoinstall ? "selected" : "not selected"}. ${lang.autoinstallInteractiveDescription}',
```

Comment:
All strings that would be read by the screen reader or by real people should be added to the corresponding English .arb file instead of directly in the code itself. 

You can find a simple example of how that works here: https://github.com/canonical/ubuntu-desktop-provision/blob/main/apps/ubuntu_bootstrap/lib/pages/done/done_page.dart#L20-L28, which has the corresponding string here: https://github.com/canonical/ubuntu-desktop-provision/blob/main/apps/ubuntu_bootstrap/lib/l10n/ubuntu_bootstrap_en.arb#L542

You can find more examples of this throughout the project.

This is applicable to more than just this line, make sure to use the same process for all the strings you add.